### DISCUSSION TITLE
Just got a new Mac (upgraded from Intel to M3) and deployments fail

### DISCUSSION THREAD

**Discussion by avisser** [aws/aws-dotnet-deploy#842] [Q&A] [answered]
As the title says, I just got a new mac with a new architecture in the chip.

I'm unable to deploy my existing stack from my new machine. The deployments take a while, then rollback. When I go to Cloudwatch, I see errors indicating that the amd64 EC2 image cannot execute `dotnet` because it's an arm image.

Testing locally it looks like all I need to do is pass `--platform linux/amd64` to `docker build`. In my testing, that builds an amd64 image on my M3 Mac. 

Can anyone share how I can do that?

**Comment by bhoradc** [comment]
Hi @avisser, 

`docker buildx build --platform=linux/amd64` should ideally work in this case. Are you still facing issue with it? If so, kindly provide details about it. 

Regards, 
Chaitanya

**Reply by avisser** [reply]
Yes, the above docker command will build my image, but I'm running `dotnet aws deploy` and it's building the docker image for me - not sure how I can inject the `--platform` arg into it.

**Reply by ashovlin** [reply, marked as answer]
All of the recipes for container-based services have a "Docker Build Args" option setting where you should be able to specify `--platform=linux/amd64`

This is available under the "more" command in the CLI which displays the advanced settings:
![image](https://github.com/user-attachments/assets/0d721b70-7d2b-40f6-bec8-b89fff553405)


Or can be specified in the [Deployment Settings File](https://aws.github.io/aws-dotnet-deploy/docs/cicd/deployment-settings-file/) if you're not deploying interactively. Here's an example we use for testing with the ECR recipe: https://github.com/aws/aws-dotnet-deploy/blob/ff86c9ca63b617bdf9a8f501d15117233ed7e6d9/test/AWS.Deploy.CLI.Common.UnitTests/ConfigFileDeployment/TestFiles/SettingsSnapshot_PushImageECR.json#L7

**Reply by avisser** [reply]
That was part of it. The other part was updating my builder image. 

`FROM mcr.microsoft.com/dotnet/aspnet:7.0 AS build` 
needed to become
`FROM mcr.microsoft.com/dotnet/sdk:7.0-bullseye-slim-arm64v8 AS build`

**Reply by avisser** [reply]
Thanks for your help!
